src/uct/api/uct.h

```diff
@@ -314,16 +314,18 @@ enum uct_am_cb_flags {
  * @brief  Memory domain capability flags.
  */
 enum {
-    UCT_MD_FLAG_ALLOC     = UCS_BIT(0),  /**< MD support memory allocation */
-    UCT_MD_FLAG_REG       = UCS_BIT(1),  /**< MD support memory registration */
-    UCT_MD_FLAG_NEED_MEMH = UCS_BIT(2),  /**< The transport needs a valid local
-                                              memory handle for zero-copy operations */
-    UCT_MD_FLAG_NEED_RKEY = UCS_BIT(3),  /**< The transport needs a valid
-                                              remote memory key for remote memory
-                                              operations */
-    UCT_MD_FLAG_ADVISE    = UCS_BIT(4),  /**< MD support memory advice */
-    UCT_MD_FLAG_FIXED     = UCS_BIT(5)   /**< MD support memory allocation with
-                                              fixed address */
+    UCT_MD_FLAG_ALLOC      = UCS_BIT(0),  /**< MD supports memory allocation */
+    UCT_MD_FLAG_REG        = UCS_BIT(1),  /**< MD supports memory registration */
+    UCT_MD_FLAG_NEED_MEMH  = UCS_BIT(2),  /**< The transport needs a valid local
+                                               memory handle for zero-copy operations */
+    UCT_MD_FLAG_NEED_RKEY  = UCS_BIT(3),  /**< The transport needs a valid
+                                               remote memory key for remote memory
+                                               operations */
+    UCT_MD_FLAG_ADVISE     = UCS_BIT(4),  /**< MD supports memory advice */
+    UCT_MD_FLAG_FIXED      = UCS_BIT(5),  /**< MD supports memory allocation with
+                                               fixed address */
+    UCT_MD_FLAG_SHARE_PTR  = UCS_BIT(6)   /**< MD supports direct access via pointer
```

Comment:
since this is UCT, I would not be super picky about this. I'm okay with UCT_MD_FLAG_RKEY_PTR even so "SHARE" is more representative